romancal/tweakreg/astrometric_utils.py

```diff
@@ -0,0 +1,186 @@
+
+"""
+
+
+def create_astrometric_catalog(input_models, catalog=DEF_CAT, output="ref_cat.ecsv",
+                               gaia_only=False, table_format="ascii.ecsv",
+                               existing_wcs=None, num_sources=None):
+    f"""Create an astrometric catalog that covers the inputs' field-of-view.
+
+    Parameters
+    ----------
+    input : str, list
+        Filenames of images to be aligned to astrometric catalog
+
+    catalog : str, optional
+        Name of catalog to extract astrometric positions for sources in the
+        input images' field-of-view. Default: {DEF_CAT}. Options available are
+        documented on the catalog web page.
+
+    output : str, optional
+        Filename to give to the astrometric catalog read in from the master
+        catalog web service.  If None, no file will be written out.
+
+    gaia_only : bool, optional
+        Specify whether or not to only use sources from GAIA in output catalog
+
+    existing_wcs : model
+        existing WCS object specified by the user as generated by
+        `resample.resample_utils.make_output_wcs`
+
+    num_sources : int
+        Maximum number of brightest/faintest sources to return in catalog.
+        If `num_sources` is negative, return that number of the faintest
+        sources.  By default, all sources are returned.
+
+    Notes
+    -----
+    This function will point to astrometric catalog web service defined
+    through the use of the ASTROMETRIC_CATALOG_URL environment variable.
+
+    Returns
+    -------
+    ref_table : `~astropy.table.Table`
+        Astropy Table object of the catalog
+
+    """
+
+    # start by creating a composite field-of-view for all inputs
+    # This default output WCS will have the same plate-scale and orientation
+    # as the first member in the list.
+    # Fortunately, for alignment, this doesn't matter since no resampling of
+    # data will be performed.
+    if existing_wcs is not None:
+        outwcs = existing_wcs
+    else:
+        outwcs = resample_utils.make_output_wcs(input_models)
+    radius, fiducial = compute_radius(outwcs)
+
+    # perform query for this field-of-view
+    ref_dict = get_catalog(fiducial[0], fiducial[1], sr=radius, catalog=catalog)
+    colnames = ('ra', 'dec', 'mag', 'objID')
+
+    ref_table = ref_dict[colnames]
+
+    # Add catalog name as meta data
+    ref_table.meta['catalog'] = catalog
+    ref_table.meta['gaia_only'] = gaia_only
+
+    # rename coordinate columns to be consistent with tweakwcs
+    ref_table.rename_column('ra', 'RA')
+    ref_table.rename_column('dec', 'DEC')
+
+    # Append GAIA ID as a new column to the table...
+    gaia_sources = []
+    for source in ref_dict:
+        if 'GAIAsourceID' in source:
+            g = source['GAIAsourceID']
+            if gaia_only and g.strip() == '':
+                continue
+        else:
+            g = "-1"  # indicator for no source ID extracted
```

Comment:
I take the `gaia_only` description in the docstring to mean that other catalogs will be tried as well. If that's the case I odn't see this handled in the code. If not, should we improve the definition? Not for this PR but just commenting here for future improvements.

Reply:
I agree with your point.
I'm sure that it'll be handled properly by the new implementation of `get_catalog`, which will use AstroQuery.
With the new implementation, it'll be easy to use `gaia_only` to switch between running a query against Gaia only or not.